ui/src/components/App.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import logo from '../img/logo.svg';
+import { Navbar, Provider } from 'rendition';
+import { NetworkInfoForm } from './NetworkInfoForm';
+import { Notifications } from './Notifications';
+
+export interface NetworkInfo {
+  ssid?: string;
+  identity?: string;
+  passphrase?: string;
+}
+
+export interface Network {
+  ssid: string;
+  security: string;
+}
+
+interface AppState {
+  isConnecting: boolean;
+  isFetchingNetworks: boolean;
+  availableNetworks: Network[];
+  error: string;
+}
+
+class App extends React.PureComponent<{}, AppState> {
+  state = {
+    isConnecting: false,
+    isFetchingNetworks: true,
+    availableNetworks: [],
+    error: ''
+  };
+
+  componentDidMount() {
+    fetch('/networks')
+      .then(data => {
+        if (data.status !== 200) {
+          throw new Error();
+        }
+
+        return data.json();
+      })
+      .then((data: Network[]) => {
+        this.setState({ availableNetworks: data });
+      })
+      .catch((e: Error) => {
+        console.log(e);
+        this.setState({ error: 'Failed to fetch available networks' });
```

Comment:
Is it worth adding the error message to this?
```suggestion
        this.setState({ error: `Failed to fetch available networks: ${error.message || error}` });
```

Reply:
I thought just logging it in the console would be enough, but sure, it is easier for users to see what the problem is.